src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -689,6 +691,31 @@ def refund(self, request, pk=None):  # pylint:disable=unused-argument
 
         return Response(status=HTTPStatus.ACCEPTED)
 
+    @extend_schema(
+        request=None,
+        responses={
+            (200, "text/csv"): OpenApiTypes.OBJECT,
+            404: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(methods=["GET"], detail=False)
+    def export(self, request):
+        """
+        Export orders to a CSV file.
+        """
+        queryset = self.filter_queryset(self.get_queryset()).order_by("created_on")
+        serializer = self.get_serializer(queryset, many=True)
```

Comment:
I think we should put an `if` statement in the case where the extract of orders contains no objects. This will avoid the `IndexError` at this line : `writer.writerow(data[0].keys())`
```suggestion
        if not queryset:
            return Http404("No order in queryset to generate .csv file..")
        serializer = self.get_serializer(queryset, many=True)
```